File: server/models/fullRound.js
var mongoose = require('mongoose'),
    Schema   = mongoose.Schema;

var fullRoundSchema = new Schema({
  letter:   { type: String },
  status:   { type: String },
  roundId:   { type: Number },
  gameId:   { type: Number },
  categories: [ {type: String} ],
  gameStatus:   { type: String },
  gameMode: {type: String}
});

fullRoundSchema.methods.fillData = function fillData(game, round) {
  this.letter = round.letter;
  this.status = round.status;
  this.roundId = round.roundId;

  this.gameId = game.gameId;
  
  this.categories=[];

  for (var i = game.categories.length - 1; i >= 0; i--) {
    this.categories.push(game.categories[i].name);
  };

  this.categories = game.categories;
  this.gameStatus = game.status;
  this.gameMode = game.mode;
}

module.exports = mongoose.model('FullRound', fullRoundSchema);

File: client/TuttiFrutti/TuttiFruttiAPI/src/main/java/com/example/FullRound.java
package com.example;

/**
 *
 * var fullRoundSchema = new Schema({
 letter:   { type: String },
 status:   { type: String },
 roundId:   { type: Number },
 gameId:   { type: Number },
 categories: [ {type: String} ],
 gameStatus:   { type: String }
 });

 */


public class FullRound {
    private String letter;
    private String status;
    private Integer roundId;
    private Integer gameId;
    private String[] categories;
    private String gameStatus;

    public void setLetter(String letter)
    {
        this.letter=letter;
    }

    public void setStatus(String status)
    {
        this.status=status;
    }

    public void setRoundId(Integer roundId)
    {
        this.roundId=roundId;
    }

    public void setGameId(Integer gameId)
    {
        this.gameId=gameId;
    }

    public void setCategories(String[] categories)
    {
        this.categories=categories;
    }

    public void setGameStatus(String gameStatus)
    {
        this.gameStatus=gameStatus;
    }

    public String getLetter()
    {
        return this.letter;
    }

    public String getStatus()
    {
        return this.status;
    }

    public Integer getRoundId()
    {
        return this.roundId;
    }

    public Integer getGameId()
    {
        return this.gameId;
    }

    public String[] getCategories()
    {
        return this.categories;
    }

    public String getGameStatus()
    {
        return this.gameStatus;
    }

}
